### PR TITLE
Implement JWT auth

### PR DESCRIPTION
**Pull request by grascm**

### CODE REVIEW

**Comment by grascm** [comment]
@jasonodonnell gentle ping 🙂

**Comment by jasonodonnell** [comment]
Thanks for the ping @grascm, we will take a look at this ASAP! 👍

**Review by jasonodonnell — CHANGES_REQUESTED** [review]
Thanks for the ping, I'm taking a look now! In general this looks good but I need to test some stuff.

Small ask: can you remove the `dist/index.js` change? We prefer to do this build on our end only when we're ready to release new versions.

File: package.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
     "homepage": "https://github.com/hashicorp/vault-action#readme",
     "dependencies": {
         "got": "^11.5.1",
-        "jsonata": "^1.8.2"
+        "jsonata": "^1.8.2",
```
Comment (jasonodonnell) [inline package.json:48]: Were these updates needed or introduced by accident? If not we'll handle these updates.

Comment (grascm) [reply package.json:48]: jsonata version remains the same as before, it's just comma in the end of the line.
jsrsasign is new library, for JWT tokens generation, that i choose from recommended libraries on https://jwt.io

Comment (jasonodonnell) [reply package.json:48]: Oh my apologies 😄 , I see now. Thanks!

File: README.md [hunk 2]
```diff
@@ -79,6 +79,17 @@ with:
...
   caCertificate: ${{ secrets.VAULTCA }}
 ```
+- **jwt**: you must provide a `role` & `privateKey` parameters, additionally you can pass `keyPassword` & `jwtTtlSec` parameters
+```yaml
+...
+with:
+  url: https://vault.mycompany.com:8200
+  method: jwt
+  role: github-action
+  privateKey: ${{ secrets.JWT_PRIVATE_KEY }}
```
Comment (jasonodonnell) [inline README.md:89]: Other [multiline parameters](https://github.com/hashicorp/vault-action#reference) in the project are base64 encoded. I think it makes sense to be consistent here with this private key and decode it when used. Thoughts?

File: action.yml [hunk 3]
```diff
@@ -53,6 +53,19 @@ inputs:
     description: 'When set to true, disables verification of the Vault server certificate.  Setting this to true in production is not recommended.'
     required: false
     default: "false"
+  role:
```
Comment (jasonodonnell) [inline action.yml:56]: Thinking these should all be prefixed with the name of the auth method as we look to add more Auth Methods.

For example:
`jwtRole`
`jwtPrivateKey`
`jwtKeyPassword`
`jwtTTL`

Thoughts?

File: action.yml [hunk 4]
```diff
@@ -53,6 +53,19 @@ inputs:
...
+    description: 'JWT auth role'
+    required: false
+  privateKey:
+    description: 'Private key to sign JWT'
+    required: false
+  keyPassword:
+    description: 'Password for key stored in privateKey (if needed)'
+    required: false
+  jwtTtlSec:
```
Comment (jasonodonnell) [inline action.yml:65]: `jwtTTL` should suffice. Seconds can be noted in description/documentation as you've done.

File: action.yml (hunk 3 above)
Comment (grascm) [reply action.yml:56]: Multiple auth methods will have `role` argument, but i can't imagine scenario, where multiple roles will be used simultaneously. So, i thought that arguments can be reusable between auth methods, if needed.

But your suggestion also valid, so i will accept you decision on this question

Comment (RichiCoder1) [reply action.yml:56]: I would almost say leave role as is and remove `JWT` from it's description. For example, the #155 PR will also feature role and there's no meaningful difference between role params for the different engines. Agree on other params though that aren't generic.

File: README.md (hunk 2 above)
Comment (grascm) [reply README.md]: Thank you for pointing out, I didn't notice that. I have fixed it now.

File: action.yml (hunk 3 above)
Comment (grascm) [reply action.yml:56]: The final version is:
`role`
`jwtPrivateKey`
`jwtKeyPassword`
`jwtTtl`

**Review by jasonodonnell — APPROVED** [review]
LGTM, thanks!

**Comment by erickufrin** [comment]
Is there a good setup document that walks through ground-up configuration of JWT auth? I too am familiar with the GitLab implementation and am looking for how I can implement GitHub actions the the same type of auth.

**Pull request merged**